testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -3929,3 +3937,505 @@ fn check_block_heights() {
+    };
+    sibling_block_header.sign_miner(&miner_sk).unwrap();
+
+    let sibling_block = NakamotoBlock {
+        header: sibling_block_header.clone(),
+        txs: vec![
+            StacksTransaction {
+                version: TransactionVersion::Testnet,
+                chain_id: 1,
+                auth: TransactionAuth::Standard(TransactionSpendingCondition::Singlesig(
+                    SinglesigSpendingCondition {
+                        hash_mode: SinglesigHashMode::P2PKH,
+                        signer: Hash160([0; 20]),
+                        nonce: 0,
+                        tx_fee: 0,
+                        key_encoding: TransactionPublicKeyEncoding::Compressed,
+                        signature: MessageSignature([0; 65]),
+                    },
+                )),
+                anchor_mode: TransactionAnchorMode::Any,
+                post_condition_mode: TransactionPostConditionMode::Allow,
+                post_conditions: vec![],
+                payload: TransactionPayload::TenureChange(TenureChangePayload {
+                    tenure_consensus_hash: sibling_block_header.consensus_hash.clone(),
+                    prev_tenure_consensus_hash: reorg_to_block.header.consensus_hash.clone(),
+                    burn_view_consensus_hash: sibling_block_header.consensus_hash.clone(),
+                    previous_tenure_end: reorg_to_block.block_id(),
+                    previous_tenure_blocks: 1,
+                    cause: stacks::chainstate::stacks::TenureChangeCause::BlockFound,
+                    pubkey_hash: Hash160::from_node_public_key(&miner_pk),
+                }),
+            },
+            last_tenure.txs[1].clone(),
+        ],
+    };
+
+    assert!(
+        !sortitions_view
+            .check_proposal(&signer_client, &signer_db, &sibling_block, &miner_pk)
+            .unwrap(),
+        "A sibling of a previously approved block must be rejected."
+    );
+
+    let start_sortition = &reorg_to_block.header.consensus_hash;
+    let stop_sortition = &sortitions_view.cur_sortition.prior_sortition;
+    // check that the get_tenure_forking_info response is sane
+    let fork_info = signer_client
+        .get_tenure_forking_info(start_sortition, stop_sortition)
+        .unwrap();
+
+    // it should start and stop with the given inputs (reversed!)
+    assert_eq!(fork_info.first().unwrap().consensus_hash, *stop_sortition);
+    assert_eq!(fork_info.last().unwrap().consensus_hash, *start_sortition);
+
+    // every step of the return should be linked to the parent
+    let mut prior: Option<&TenureForkingInfo> = None;
+    for step in fork_info.iter().rev() {
+        if let Some(ref prior) = prior {
+            assert_eq!(prior.sortition_id, step.parent_sortition_id);
+        }
+        prior = Some(step);
+    }
+
+    // view is stale, if we ever expand this test, sortitions_view should
+    // be fetched again, so drop it here.
+    drop(sortitions_view);
+
```

Comment:
Did you want to add a test to see what happens if there's a Bitcoin reorg?  Or is that a separate PR?

Reply:
I'm not really sure what the impact of bitcoin forks is on this PR. This PR depends on sortition state, querying existing interfaces for sortition information, keyed by consensus hashes. It's true that bitcoin forks are handled by the sortition db, but this PR is really a client of that work. The question of bitcoin fork's impacts on the system is really an end-to-end question about the block processing and sortition db, which I'd really resist tieing this PR to.

Reply:
Yeah, that's my feeling as well.  However, before release, we'll want to extensively test the system with Bitcoin reorgs to be sure that the signers all correctly track the canonical fork at all times (even if it flaps)